core/organization/service.go

```diff
@@ -81,7 +82,11 @@ func (s Service) Update(ctx context.Context, org Organization) (Organization, er
 func (s Service) AddAdmins(ctx context.Context, idOrSlug string, userIds []string) ([]user.User, error) {
 	currentUser, err := s.userService.FetchCurrentUser(ctx)
 	if err != nil {
-		return []user.User{}, err
+		return []user.User{}, fmt.Errorf("%w: %s", user.ErrInvalidEmail, err.Error())
```

Comment:
in lot of places we are returning `[]user.User{}`. Wherever there's a need to use empty slice, we should be using `nil`. It prevents an allocation. And has no difference (It cannot cause any nil pointer issues).

⚠️ Only exception to this rule is during JSON marshal. Because `nil` slice value is written out as `null` and empty-slice is marshaled as `[]` (json array)

Reply:
@spy16 Agree on this, we could just return nil for the empty slices. I keep that part apart from the refactor and this PR just want to don't want to have more-huge changes on this. 

This could be addressed in separate PR and issue.

Reply:
makes sense.